sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfigTest.java

```diff
@@ -0,0 +1,238 @@
+    assertEquals(CONFIGURATOR,
+      config.withBigtableOptionsConfigurator(CONFIGURATOR).getBigtableOptionsConfigurator());
+
+    thrown.expect(IllegalArgumentException.class);
+    config.withBigtableOptionsConfigurator(null);
+  }
+
+  @Test
+  public void testWithValidate() {
+    assertEquals(true, config.withValidate(true).getValidate());
+  }
+
+  @Test
+  public void testWithBigtableService() {
+    assertEquals(SERVICE, config.withBigtableService(SERVICE).getBigtableService());
+
+    thrown.expect(IllegalArgumentException.class);
+    config.withProjectId(null);
+  }
+
+  @Test
+  public void testValidate() {
+    config.withProjectId(PROJECT_ID)
+      .withInstanceId(INSTANCE_ID)
+      .withTableId(TABLE_ID)
+      .validate();
+  }
+
+  @Test
+  public void testValidateFailsWithoutProjectId() {
+    config.withInstanceId(INSTANCE_ID)
+      .withTableId(TABLE_ID);
+
+    thrown.expect(IllegalArgumentException.class);
+    config.validate();
+  }
+
+  @Test
+  public void testValidateFailsWithoutInstanceId() {
+    config.withProjectId(PROJECT_ID)
+      .withTableId(TABLE_ID);
+
+    thrown.expect(IllegalArgumentException.class);
+    config.validate();
+  }
+
+  @Test
+  public void testValidateFailsWithoutTableId() {
+    config.withProjectId(PROJECT_ID)
+      .withInstanceId(INSTANCE_ID);
+
+    thrown.expect(IllegalArgumentException.class);
+    config.validate();
+  }
+
+  @Test
+  public void testPopulateDisplayData() {
+    DisplayData displayData = DisplayData.from(new HasDisplayData() {
+      @Override
+      public void populateDisplayData(DisplayData.Builder builder) {
+        config.withProjectId(PROJECT_ID)
+          .withInstanceId(INSTANCE_ID)
+          .withTableId(TABLE_ID)
+          .populateDisplayData(builder);
+      }
+    });
+
+    assertThat(displayData, hasDisplayItem(allOf(
+      hasKey("projectId"),
+      hasLabel("Bigtable Project Id"),
+      hasValue(PROJECT_ID.get()))));
+
+    assertThat(displayData, hasDisplayItem(allOf(
+      hasKey("instanceId"),
+      hasLabel("Bigtable Instance Id"),
+      hasValue(INSTANCE_ID.get()))));
+
+    assertThat(displayData, hasDisplayItem(allOf(
+      hasKey("tableId"),
+      hasLabel("Bigtable Table Id"),
+      hasValue(TABLE_ID.get()))));
+  }
+
+  @Test
+  public void testGetBigtableServiceWithDefaultService() {
+    assertEquals(SERVICE, config.withBigtableService(SERVICE).getBigtableService());
+  }
+
+  @Test
+  public void testGetBigtableServiceWithoutConfigurator() {
+    GcpOptions pipelineOptions = PipelineOptionsFactory.as(GcpOptions.class);
+
+    BigtableService service = config
+      .withProjectId(PROJECT_ID)
+      .withInstanceId(INSTANCE_ID)
+      .getBigtableService(pipelineOptions);
+
+    assertEquals(PROJECT_ID.get(), service.getBigtableOptions().getProjectId());
+    assertEquals(INSTANCE_ID.get(), service.getBigtableOptions().getInstanceId());
+    assertEquals(CredentialOptions.CredentialType.SuppliedCredentials,
```

Comment:
This check can only run on machines that have Google application default credentials configured.

Reply:
Addressed